composer/trainer/devices/device.py

```diff
@@ -123,3 +117,24 @@ def precision_context(self, precision: Union[str, Precision]) -> Generator[None,
             Generator[None, None, None]: A context for the precision.
         """
         pass
+
+
+def _map_batch(batch: Any, map_fn: Callable) -> Any:
+    """Recursively maps a function to all items in a batch.
+
+    Args:
+        batch: Nested lists and dictionaries.
+        map_fn: A function to invoke on each element.
+
+    Returns:
+        Collections: The result of applying ``map_fn`` on each element of the ``batch``.
+        The type of ``batch`` is preserved.
+    """
+    if isinstance(batch, Mapping):
+        return {k: _map_batch(v, map_fn) for k, v in batch.items()}
```

Comment:
I can't think of many mapping objects that support a `type()` other than converting TO `dict` style conversion but it can't hurt. try: except: is cheap.